vispy/gloo/glir.py

```diff
@@ -1320,7 +1321,16 @@ def draw(self, mode, selection):
             raise RuntimeError('Cannot draw program if code has not been set')
         # Init
         gl.check_error('Check before draw')
-        mode = as_enum(mode)
+        try:
+            mode = as_enum(mode)
+        except ValueError as err:
+            if mode == 'lines_adjacency' or mode == 'line_strip_adjacency':
+                raise RuntimeError(gl.current_backend.__name__ + " backend does"
+                " not support lines_adjacency and line_strip_adjacency"
+                " primitives. Try gloo.gl.use_gl('gl+').")
+            else:
+                raise err
```

Comment:
Could you remove the `as err` part and change this to `raise`, that way the original traceback is preserved.

Reply:
Actually you can remove the `else:` block and just do `raise`:

```python
except ValueError:
    if mode ...:
        raise RuntimeError(...)
    raise
```